src/thor/isochrone.cc

```diff
@@ -200,76 +245,59 @@ void Isochrone::UpdateIsoTile(const EdgeLabel& pred,
   }
 
   // Get the time and distance at the end node of the predecessor
-  // TODO - do we need partial shape from origin location to end of edge?
   float secs1 = pred.cost().secs;
   float dist1 = static_cast<float>(pred.path_distance());
 
   // For short edges just mark the segment between the 2 nodes of the edge. This
   // avoid getting the shape for short edges.
-  if (edge->length() < shape_interval_ * 1.5f) {
-    // Mark tiles that intersect the segment. Optimize this to avoid calling the Intersect
-    // method unless more than 2 tiles are crossed by the segment.
+  auto len = pred.origin() ? pred.path_distance() : edge->length();
+  if (len < shape_interval_ * 1.5f) {
     PointLL ll0 = tile->get_node_ll(t2->directededge(opp)->endnode());
-    auto tile1 = isotile_->TileId(ll0);
-    auto tile2 = isotile_->TileId(ll);
-    if (tile1 == tile2) {
-      isotile_->SetIfLessThan(tile1, {secs1 * kMinPerSec, dist1 * kKmPerMeter});
-    } else if (isotile_->AreNeighbors(tile1, tile2)) {
-      // If tile 2 is directly east, west, north, or south of tile 1 then the
-      // segment will not intersect any other tiles other than tile1 and tile2.
-      isotile_->SetIfLessThan(tile1, {secs1 * kMinPerSec, dist1 * kKmPerMeter});
-      isotile_->SetIfLessThan(tile2, {secs1 * kMinPerSec, dist1 * kKmPerMeter});
-    } else {
-      // Find intersecting tiles (using a Bresenham method)
-      auto tiles = isotile_->Intersect(std::list<PointLL>{ll0, ll});
-      for (const auto& t : tiles) {
-        isotile_->SetIfLessThan(t.first, {secs1 * kMinPerSec, dist1 * kKmPerMeter});
-      }
+    if (pred.origin()) {
+      // interpolate ll0 for origin edge using edge_label.path_distance()
+      auto edge_info = tile->edgeinfo(edge);
+      const auto& shape = edge_info.shape();
+      const auto& ordered_shape =
+          edge->forward() ? shape : std::vector<midgard::PointLL>(shape.rbegin(), shape.rend());
+      auto origin_edge_shape = OriginEdgeShape(ordered_shape, pred.path_distance());
```

Comment:
a reference to a temporary, does that work? maybe to be safe:

```suggestion
      if(!edge->forward):
        std::reverse(shape.begin(), shape.end());
      auto origin_edge_shape = OriginEdgeShape(shape, pred.path_distance());
```

Reply:
this is the trick to avoid shape copying if reverse is not needed https://abseil.io/tips/107

Reply:
Ah I see constifying it extended the lifetime. That makes sense I use this trick a lot for default parameter values from temporaries